File: scripts/FileChooser.lua
-- FileChooser.lua, Example showing how to use a file chooser in Lua
-- This is meant to be run as a script within Supremica
-- This is the example given in the Lupremica paper (IFAC WC 2023)
local luaj = luajava -- just shorthand
local script, ide, log = ... -- parameters from Supremica

local fc = luaj.newInstance("javax.swing.JFileChooser")
fc:setDialogTitle("Lupremica file chooser")
local retval = fc:showOpenDialog()
if retval == fc.APPROVE_OPTION then
	local fname = fc:getSelectedFile():getPath() -- does not work on Java > 8
	-- local fname = fc:getName(fc:getSelectedFile()) -- this works for Java > 8
	print("File: "..fname)
else
	print("User cancelled")
end

File: scripts/FileChooser.java
/********************* FileChooser.java *******************/
/* Trying the FileChooser.lua script as Java code. There is
 * a bug in either LuaJava or in Java itself that prevents
 * the simple call fc:getSelectedFile():getName(), or variants
 * of it, from working; an exception is thrown instead
 ** It works fine from Java, though
 */
package Lupremica;

import java.io.File;
import javax.swing.JFileChooser;
import org.supremica.gui.ide.IDE;
import org.apache.logging.log4j.Logger;

public class FileChooser
{

    public FileChooser(final IDE ide)
    {
		final Logger log = ide.getTheLog();

		final JFileChooser fc = new JFileChooser();
		fc.setDialogTitle("Lupremica file chooser");
		final int retval = fc.showOpenDialog(null);
		if(retval == JFileChooser.APPROVE_OPTION)
		{
			log.info("File: " + fc.getSelectedFile().getName());
		}
		else
		{
			log.info("User cancelled");
		}
    }
}